src/stratigraphy/main.py

```diff
@@ -225,12 +225,18 @@ def start_pipeline(
                 # Extract metadata
                 metadata = BoreholeMetadata.from_document(doc)
 
+                # Save the predictions to the overall predictions object
+                # Initialize common variables
+                groundwater_entries = None
+                layers = None
+                depths_materials_columns_pairs = None
+
                 if part == "all":
                     # Extract the groundwater levels
-                    groundwater_extractor = GroundwaterLevelExtractor(document=doc)
-                    groundwater = groundwater_extractor.extract_groundwater(terrain_elevation=metadata.elevation)
+                    groundwater_in_document = GroundwaterInDocument.from_document(doc, metadata.elevation)
```

Comment:
Directly assign the result to `groundwater_entries`, instead of using this redundant intermediate variable `groundwater_in_document`, and copying the value later on.

Same for `layers` and `depths_materials_columns_pairs`.

Reply:
Makes sense!